modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CategoriesListRowRedesignAdapter.kt

```diff
@@ -0,0 +1,127 @@
+package au.com.shiftyjelly.pocketcasts.discover.view
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat.getDrawable
+import androidx.core.content.ContextCompat.getString
+import androidx.recyclerview.widget.RecyclerView
+import au.com.shiftyjelly.pocketcasts.discover.R
+import au.com.shiftyjelly.pocketcasts.discover.databinding.CategoryPillBinding
+import au.com.shiftyjelly.pocketcasts.servers.model.DiscoverCategory
+import au.com.shiftyjelly.pocketcasts.servers.model.NetworkLoadableList
+
+@SuppressLint("NotifyDataSetChanged")
+class CategoriesListRowRedesignAdapter(
+    private val onCategoryClick: (NetworkLoadableList) -> Unit,
+    private val onAllCategoriesClick: (() -> Unit) -> Unit,
+    private val categories: List<CategoryPill>,
+) : RecyclerView.Adapter<CategoriesListRowRedesignAdapter.CategoriesRedesignViewHolder>() {
+    companion object {
+        const val ALL_CATEGORIES_INDEX = 0
+    }
+
+    private val currentCategories = categories.map { it.copy() }.toMutableList()
+
+    class CategoriesRedesignViewHolder(
+        val binding: CategoryPillBinding,
+        private val onItemClicked: (Int) -> Unit,
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        init {
+            binding.categoryPill.setOnClickListener {
+                onItemClicked(bindingAdapterPosition)
+            }
+        }
+        fun bind(category: CategoryPill, context: Context) {
+            if (category.discoverCategory.id == DiscoverCategory.ALL_CATEGORIES_ID) {
+                setUpAllCategoriesAndClear(context, category)
+            } else {
+                setUpCategories(context, category)
+            }
+        }
+        private fun setUpAllCategoriesAndClear(context: Context, category: CategoryPill) {
+            if (category.isSelected) {
+                binding.categoryName.visibility = View.GONE
```

Comment:
Do you think we should use `isVisibility = false` here for better readable code?

Reply:
Yeah. Looks better with `isVisibility `. I've just updated the code